infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -0,0 +1,204 @@
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.util.Utils;
+
+import javax.annotation.Nonnull;
+import java.util.UUID;
+
+/**
+ * Sequencer server cache.
+ * Contains transaction conflict-resolution data structures.
+ *
+ * The SequencerServer use its own thread/s. To guarantee correct tx conflict-resolution,
+ * the {@link SequencerServerCache#conflictCache} must be updated
+ * along with {@link SequencerServerCache#maxConflictWildcard} at the same time (atomically) to prevent race condition
+ * when the conflict stream is already evicted from the cache but `maxConflictWildcard` is not updated yet,
+ * which can cause situation when sequencer let the transaction go but the tx has to be cancelled.
+ */
+@Slf4j
+public class SequencerServerCache {
+    /**
+     * TX conflict-resolution information:
+     * <p>
+     * a cache of recent conflict keys and their latest global-log position.
+     */
+    private final Cache<ConflictTxStream, Long> conflictCache;
+
+    /**
+     * A "wildcard" representing the maximal update timestamp of
+     * all the conflict keys which were evicted from the cache
+     */
+    @Getter
+    private long maxConflictWildcard = Address.NOT_FOUND;
+
+    /**
+     * maxConflictNewSequencer represents the max update timestamp of all the conflict keys
+     * which were evicted from the cache by the time this server is elected
+     * the primary sequencer. This means that any snapshot timestamp below this
+     * actual threshold would abort due to NEW_SEQUENCER cause.
+     */
+    @Getter
+    private long maxConflictNewSequencer = Address.NOT_FOUND;
+
+    @VisibleForTesting
+    public SequencerServerCache(long cacheSize, CacheWriter<ConflictTxStream, Long> writer) {
+        this.conflictCache = Caffeine.newBuilder()
+                .maximumSize(cacheSize)
+                .writer(writer)
+                //Performing periodic maintenance using current thread (synchronously)
+                .executor(Runnable::run)
+                .recordStats()
+                .build();
+    }
+
+    /**
+     * The cache limited by size.
+     * For a synchronous cache we are using a same-thread executor (Runnable::run)
+     * https://github.com/ben-manes/caffeine/issues/90
+     *
+     * @param cacheSize cache size
+     */
+    public SequencerServerCache(long cacheSize) {
+        this.conflictCache = Caffeine.newBuilder()
+                .maximumSize(cacheSize)
+                //Performing periodic maintenance using current thread (synchronously)
+                .executor(Runnable::run)
+                .writer(getDefaultCacheWriter())
+                .recordStats()
+                .build();
+    }
+
+    private CacheWriter<ConflictTxStream, Long> getDefaultCacheWriter() {
+        return new CacheWriter<ConflictTxStream, Long>() {
+            /**
+             * Let caffeine do its work
```

Comment:
Can you rephrase this generic comment?

Reply:
done.
Just FYI: Also, caffeine has very good documented methods, it should be clear from caffeine java doc